client/scripts/controllers/chain/edgeware/signaling_proposal.ts

```diff
@@ -0,0 +1,249 @@
+import { ApiRx } from '@polkadot/api';
+import { Option, Vec } from '@polkadot/types';
+import { ITuple } from '@polkadot/types/types';
+import { AccountId } from '@polkadot/types/interfaces';
+import { VoteRecord } from 'edgeware-node-types/dist/types';
+import { IEdgewareSignalingProposal } from 'adapters/chain/edgeware/types';
+import {
+  Account, Proposal, ProposalStatus, ProposalEndTime, IVote, VotingType,
+  VotingUnit, ChainClass, ChainEntity, ChainEvent
+} from 'models';
+import { default as SubstrateChain } from 'controllers/chain/substrate/shared';
+import SubstrateAccounts, { SubstrateAccount } from 'controllers/chain/substrate/account';
+import { BehaviorSubject, Unsubscribable, combineLatest, of } from 'rxjs';
+import { SubstrateCoin } from 'adapters/chain/substrate/types';
+import { ISubstrateSignalingNewProposal, SubstrateEventKind } from 'events/edgeware/types';
+import { VoteOutcome } from 'edgeware-node-types/dist';
+import EdgewareSignaling from './signaling';
+
+export enum SignalingProposalStage {
+  PreVoting = 'prevoting',
+  Commit = 'commit',
+  Voting = 'voting',
+  Completed = 'completed',
+}
+
+const backportEventToAdapter = (
+  ChainInfo: SubstrateChain,
+  event: ISubstrateSignalingNewProposal,
+): IEdgewareSignalingProposal => {
+  return {
+    identifier: event.proposalHash,
+    voteIndex: +event.voteId,
+    hash: event.proposalHash,
+    author: event.proposer,
+    title: event.title,
+    description: event.description,
+    tallyType: ChainInfo.createType('TallyType', event.tallyType),
+    voteType: ChainInfo.createType('VoteType', event.voteType),
+    choices: event.choices.map((c) => ChainInfo.createType('VoteOutcome', c)),
+  };
+};
+
+export class SignalingVote implements IVote<SubstrateCoin> {
+  public readonly account: SubstrateAccount;
+  public readonly choices: VoteOutcome[];
+  public readonly balance: SubstrateCoin;
+
+  constructor(
+    proposal: EdgewareSignalingProposal,
+    account: SubstrateAccount,
+    choices: VoteOutcome[],
+    balance: SubstrateCoin
+  ) {
+    this.account = account;
+    this.choices = choices;
+    this.balance = balance;
+  }
+}
+
+export class EdgewareSignalingProposal
+  extends Proposal<ApiRx, SubstrateCoin, IEdgewareSignalingProposal, SignalingVote> {
+  public get shortIdentifier() {
+    return `#${this.data.voteIndex.toString()}`;
+  }
+  public get title() { return this.data.title; }
+  public get description() { return this.data.description; }
+  public get author() { return this.data.author ? this._Accounts.fromAddress(this.data.author) : null; }
+
+  public get support() {
+    // TODO: support multi-option votes
+    // TODO: support ranked-choice votes
+    // TODO: support 1p1v
+    if (this.getVotes().some((v) => v.balance === undefined)) {
+      // balances haven't resolved yet!
+      // console.error('Balances haven\'t resolved');
+      return null;
+    }
+    const yesVotes = this.getVotes()
```

Comment:
this is where my comment is in reference to. We should also be checking for a variety of other 'yes' and 'no' like outcomes to ensure we actually catch the intended vote choice. Right now this implementation leads to 0% support for yes on the UI.

Reply:
Will take a look and fix this up, ty for pointing it out -- i may have already fixed this in the changes made since this comment.